### PR TITLE
Add connbuf unblocking convenience macro

### PR DESCRIPTION
**Pull request by neithernut**
This PR enhances the readability of the connector's code by introducing a convenience macro, as suggested by @jonashaag.
Please do not merge right away but let me do the merge.

### CODE REVIEW

**Comment by jonashaag** [comment]
Btw this doesn't need to be a macro since compilers can optimize this away easily.

The advantage of not using a macro here is that external code using that module doesn't need to be recompiled when the internal implementation of `ws_*` changes.

**Comment by neithernut** [comment]
Yes, it can be optimized _if_ we use LTO.
I'll change it anyways. We can then rewrite `ws_connbuf_append()` to use `ws_connbuf_unblock()`, which _kind of_ makes sense.

**Comment by matthiasbeyer** [comment]
@jonashaag Are you fine with a "Suggested-by:" tag in the commit message?

File: src/connection/connbuf.c [hunk 1]
```diff
@@ -104,8 +104,14 @@ ws_connbuf_append(
     }
 
     self->data += amount;
-    self->blocked = false;
+    return ws_connbuf_unblock(self);
+}
 
+int
+ws_connbuf_unblock(
+    struct ws_connbuf* self //!< The object
```
Comment (matthiasbeyer) [inline src/connection/connbuf.c:112]: You can remove the comment on this line!

**Comment by matthiasbeyer** [comment]
Besides the one annotation: ACK. Smart PR!

Wait for an answer from @jonashaag , though!

**Comment by jonashaag** [comment]
I don't care

**Comment by neithernut** [comment]
In this case, the PR is ready for merge.

**Comment by matthiasbeyer** [comment]
Feel free to do so.

**Pull request merged**